### PR TITLE
Add negation operators to V2D and V3D

### LINKED ISSUE #19958: V2D and V3D are missing negate operators
The negation `V?D operator-() const;` methods and Python exports could be added to `V2D` and `V3D` classes.

### PR DESCRIPTION
**Pull request by soininen**
This PR adds the unary `operator-()` functions to `V2D` and `V3D`. The negation operator of `V3D` is also exposed to Python (`V2D` itself in not available in Python).

**To test:**

Check that negation of `V3D` objects produce vectors of the same length but in opposite direction. Testing `V2D` might be a bit tricky as it is not available in Python, but since it's not used too much, maybe it's enough to do a throughout code review.

Fixes #19958.

**Release Notes** 

Framework release notes were updated accordingly.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Kernel/inc/MantidKernel/V2D.h [hunk 1]
```diff
@@ -76,6 +76,8 @@ class MANTID_KERNEL_DLL V2D {
   V2D operator*(const double factor) const;
   /// Scale this
   V2D &operator*=(const double factor);
+  /// Negate
+  V2D operator-() const noexcept;
```
Comment (josephframsay) [inline Framework/Kernel/inc/MantidKernel/V2D.h]: Can this safely be `noexcept`? For instance, is it OK to assume that `-NaN` or `-inf` won't throw anything?

**Review by josephframsay — COMMENTED** [review]
These changes are nice, and the new operator behaves as expected on my machine. Just one small query about `noexcept`, but apart from that everything looks good!

File: Framework/Kernel/inc/MantidKernel/V2D.h (hunk 1 above)
Comment (soininen) [reply Framework/Kernel/inc/MantidKernel/V2D.h]: Yes. I've added unit tests to make sure the behavior won't change unnoticed.

Comment (josephframsay) [reply Framework/Kernel/inc/MantidKernel/V2D.h]: Cool - once the build servers pass this looks good to go from my point of view then

**Review by josephframsay — APPROVED** [review]

File: Framework/Kernel/test/V2DTest.h [hunk 2]
```diff
@@ -107,6 +107,19 @@ class V2DTest : public CxxTest::TestSuite {
...
   }
 
+  void test_Negate_Gives_Same_Length_But_Opposite_Direction() {
+    const V2D p1(-3, 9);
+    const V2D p2 = -p1;
+    TS_ASSERT_EQUALS(p2, V2D(3, -9))
+  }
+
+  void test_Negate_Works_With_Special_Values() {
+    const V2D p1(INFINITY, std::nan(""));
```
Comment (quantumsteve) [inline Framework/Kernel/test/V2DTest.h]: `std::nan("")` is correct (no need to change it), but I find [`std::numeric_limits<double>::quiet_NaN()`](http://en.cppreference.com/w/cpp/types/numeric_limits/quiet_NaN) is a bit clearer.

Comment (soininen) [reply Framework/Kernel/test/V2DTest.h]: I didn't know this exists in `std::numeric_limits`. Thanks for the hint!

**Pull request merged**